browser/brave_ads/ads_service_delegate.h

```diff
@@ -78,13 +77,13 @@ class AdsServiceDelegate : public AdsService::Delegate {
   base::Value::Dict GetVirtualPrefs() override;
 
  private:
+  NotificationDisplayService* GetNotificationDisplayService();
+
   raw_ptr<Profile> profile_ = nullptr;
   raw_ptr<PrefService> local_state_ = nullptr;
   search_engines::SearchEngineChoiceService search_engine_choice_service_;
   raw_ptr<brave_adaptive_captcha::BraveAdaptiveCaptchaService>
       adaptive_captcha_service_ = nullptr;
-  raw_ptr<NotificationDisplayService, DanglingUntriaged>
-      notification_display_service_ = nullptr;
```

Comment:
I'm a bit confused by this. `notification_display_service_` is a member of `AdsServiceDelegate`, which is owned by `AdsService`. If we look in `AdsServiceFactory::AdsServiceFactory`, I can see this:

```
AdsServiceFactory::AdsServiceFactory()
    : BrowserContextKeyedServiceFactory(
          "AdsService",
          BrowserContextDependencyManager::GetInstance()) {
  DependsOn(NotificationDisplayServiceFactory::GetInstance());
```

So, `AdsService` must outlive `NotificationDisplayService`, and therefore so should `AdsServiceDelegate`. It looks like that for whatever reason this was dangling, it should not have been dangling. I'm just wondering if removing the member is the best solution here, or if there's something else going on.

Reply:
Actually `AdsService` depends on `NotificationDisplayService`, so `NotificationDisplayService` should be outlived by `AdsService`.

Reply:
I have investigated it further and this is an upstream problem... your solution is correct.

Reply:
Thanks for investigating. Much appreciated